### PR TITLE
Add history-clear command

### PR DESCRIPTION
**Pull request by fapfaff**
Added command to remove all notifications from history.

### CODE REVIEW

**Comment by fwsmit** [comment]
Just curious, what do you use this for?

**Comment by fapfaff** [comment]
For an eww widget displaying my notification history

**Comment by fapfaff** [comment]
Do you have an idea why it doesn't build?
Shouldn't g_queue_clear_full() be included by glib.h?

Sorry for asking you my noob questions, but it's working locally without any problems and I couldn't find anything that helps me.

**Comment by fwsmit** [comment]
I'm guessing the version of glib for these distro's is too old. You need at least version 2.6 to get this function. I've ran the archlinux build again and it works. This seems to confirm my suspicion. vTake a look at https://docs.gtk.org/glib/struct.Queue.html for the version requirements of different functions.

You can work around this by using other queue functions in glib.

**Comment by fapfaff** [comment]
Thanks, I have replaced it. Should work now.

File: src/queues.c [hunk 1]
```diff
@@ -347,7 +347,10 @@ void queues_history_clear(void)
         if (g_queue_is_empty(history))
```
Comment (fwsmit) [inline src/queues.c:347]: This is unnecesary

File: src/queues.c [hunk 2]
```diff
@@ -347,7 +347,10 @@ void queues_history_clear(void)
...
                 return;
 
-        g_queue_clear_full(history, (GDestroyNotify)notification_unref);
+        while (!g_queue_is_empty(history)) {
+                struct notification *n = g_queue_pop_head(history);
+                notification_unref(n);
+        }
```
Comment (fwsmit) [inline src/queues.c:353]: You can actually use the source of glib to copy the function here. See https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gqueue.c#L166

```
g_queue_foreach (queue, (GFunc) free_func, NULL);
g_queue_clear (queue);
```

**Review by fwsmit — CHANGES_REQUESTED** [review]

**Comment by fwsmit** [comment]
LGTM, thanks!

**Pull request merged**